packages/@magic-sdk/react-native-expo/src/react-native-sdk-base.ts

```diff
@@ -1,7 +1,72 @@
-import { SDKBase } from '@magic-sdk/provider';
+import { createPromiEvent, MagicSDKAdditionalConfiguration, SDKBase, UserModule } from '@magic-sdk/provider';
+import AsyncStorage from '@react-native-async-storage/async-storage';
 import { ReactNativeWebViewController } from './react-native-webview-controller';
 
+type UserLoggedOutCallback = (loggedOut: boolean) => void;
+
 export class SDKBaseReactNative extends SDKBase {
+  // this UserModule instance used for the actual calls to
+  // `isLoggedIn` and `logOut`, given that we assign new functions
+  // to `this.user.isLoggedIn` and `this.user.logOut`.
+  private usr: UserModule;
+  private userLoggedOutCallbacks: UserLoggedOutCallback[] = [];
+
+  constructor(public readonly apiKey: string, options?: MagicSDKAdditionalConfiguration) {
+    super(apiKey, options);
+    this.usr = new UserModule(this);
+    if (options?.useStorageCacheMobile) {
+      this.user.isLoggedIn = () => {
+        const promiEvent = createPromiEvent<boolean, any>(async (resolve, reject) => {
+          const isLoggedInStored = (await AsyncStorage.getItem('isLoggedIn')) === 'true';
+
+          // if isLoggedIn is true on storage, optimistically resolve with true
+          // if it is false, we use `usr.isLoggedIn` as the source of truth.
+          if (isLoggedInStored) {
+            resolve(true);
+          }
+
+          this.usr.isLoggedIn().then(async (actualValue: boolean) => {
+            if (actualValue === false) {
+              AsyncStorage.removeItem('isLoggedIn');
+              if (isLoggedInStored) {
+                this.emitUserLoggedOut(true);
+              }
+            } else {
+              console.log('saving item');
+              AsyncStorage.setItem('isLoggedIn', 'true');
```

Comment:
Is is the idea that we'd never store a `false` value?

Reply:
Yes, we just clear `isLoggedIn` from the storage. That is because we won't confidently know when the user is logged in again until the next `isLoggedIn` call.